lib/Common/Memory/WriteBarrierMacros.h

```diff
@@ -17,6 +17,8 @@
 
 #define FieldWithBarrier(type, ...) \
     typename WriteBarrierFieldTypeTraits<type, ##__VA_ARGS__>::Type
+#define FieldWithBarrierT(type, ...) \
```

Comment:
I think we should explicitly adding a typename here for vs2013 then, like 
```
#if _MSC_VER<XXX
#define TYPENAME typename
#endif
        TYPENAME FieldWithBarrier(Field(T*)*) inlineCaches;
```

and @jianchun found we already macro _TYPENAME to handle similar case.